app/models/rent.rb

```diff
@@ -3,7 +3,31 @@ class Rent < ApplicationRecord
   belongs_to :book
 
   validates :user_id, :book_id, :start_date, :end_date, presence: true
+  validate :book_availability
 
   scope :rented_by, ->(user_id) { where(user_id: user_id) }
   scope :ending_today, -> { where(end_date: Time.zone.today) }
+
+  private
+
+  def book_availability
+    return if book_id.blank? || start_date.blank? || end_date.blank?
```

Comment:
All of this check are validates by `presence: true` just above, aren't they?

Reply:
If I remove that line I could get an error when querying for a rent that overlaps with the selected time frame